### ISSUE TITLE
Avellaneda - Orders are not created immediately afterwards when max order age is reached

### ISSUE THREAD

**Issue by rxlxrxsx** [CoinAlpha/hummingbot#3808]
**Describe the bug**
When running the avellaneda market making strategy, orders are not created immediately afterwards when max order age is reached. This only happens when order refresh tolerance is enabled and the orders are retained (not canceled) because it didn't reach the value set on `order_refresh_tolerance_pct`. This issue doesn't affect hanging orders that reaches the max order age and other strategies as well.

<img width="1424" alt="Screen Shot 2021-07-29 at 2 09 30 PM" src="https://user-images.githubusercontent.com/64377055/127443608-0b8fdc75-f191-4f6c-bb31-4e14715d1ea9.png">

**Steps To Reproduce**
1. Create and run an avellaneda market making strategy.
2. On my test, I set the order_refresh_time to 30, max_order_age to 60 and order_refresh_tolerance_pct to 1 to easily reproduce the issue.
3. When there are no spread changes above or below 1%, orders are retained and max order age will cancel the orders, however it will not create new set of orders immediately and will wait until the next cycle before placing new set of orders.

**Release version**
dev-0.42.0

**Attachments**
Logs and config: [avellaneda.zip](https://github.com/CoinAlpha/hummingbot/files/6898467/avellaneda.zip)

**Comment by RC-13** [comment]
Same case is happening on PMM strategy:
![image](https://user-images.githubusercontent.com/57189990/133953732-debbd98f-3a6d-43e6-8624-c33870d93b7e.png)

https://user-images.githubusercontent.com/57189990/133953822-a8d428d5-b733-4047-865f-3e25a04eccb5.mov

**Comment by dennisocana** [comment]
@RC-13 were there no new orders created at all? If there were, how long was the "gap" in between the cycle?

**Comment by RC-13** [comment]
@RC-13 yes, no order are created in the next 30 seconds. The bot will create a new sets of order after the 60 seconds refresh time.

**Comment by nikspz** [comment]
Issue still showing for Avellaneda MM and Pure MM, staging v1.2.0

**Comment by nikspz** [comment]
(created after order_refresh_time.)
